### ISSUE TITLE
Use of uninitialized value $_ when undefined path in $PATH 

### ISSUE THREAD

**Issue by burgerdev** [olibre/colorgcc#3]
I get this warning/error message when running a simple file via colorgcc:

```text
Use of uninitialized value $_ in hash element at ./gcc line 272.
Use of uninitialized value $_ in hash element at ./gcc line 272.
Use of uninitialized value $_ in hash element at ./gcc line 272.
Use of uninitialized value $_ in hash element at ./gcc line 272.
Use of uninitialized value $_ in hash element at ./gcc line 272.
Use of uninitialized value $_ in hash element at ./gcc line 272.
Use of uninitialized value $_ in hash element at ./gcc line 272.
Use of uninitialized value $_ in string ne at ./gcc line 307.
Use of uninitialized value $_ in -x at ./gcc line 276.
Use of uninitialized value $_ in -x at ./gcc line 278.
```

File contents:

```c
int main(int c, char** argv) 
{
    return 0;
}
```

Steps to reproduce:

```bash
git clone https://github.com/olibre/colorgcc.git
cd colorgcc
ln -s colorgcc.pl gcc
PATH=.:$PATH
gcc /tmp/test.c
```

System: 

Arch Linux x86_64
Perl 5v20

**Comment by olibre** [comment]
Thanks for your bug report. :+1: 
I will check this week-end. Cheers :-)

**Comment by hzy199411** [comment]
I also get this error,when will it fixed?

**Comment by olibre** [comment]
@burgerdev 
Thank you for your bug report.
I was able to reproduce it at work (Red Hat 6).
I am at home on Ubuntu 14.10 with free time to investigate, but I cannot reproduce the issue !!!

**Comment by olibre** [comment]
Hi @burgerdev and @hzy199411 

I am sorry I am not able to reproduce the issue, even at office :(
Please provide more information as Operating System, versions...

Cheers,
Olibre

**Comment by hzy199411** [comment]
Hi, @olibre , it seems ther perl's problem, After I upgrade perl to 5.20.2, I can't reproduce the issue too.

**Comment by olibre** [comment]
Thank you @hzy199411 
I close the issue untill this can be reproduced again

**Comment by hzy199411** [comment]
@burgerdev and @olibre, Today I have reproduced this issue and I found a solution of this problem, it's not the perl's problem, it's the $PATH environment variable's problem, if one directory in your $PATH does not exist, you would get some warning like this:
`Use of uninitialized value $_ in hash element at /usr/bin/colorgcc line 272.`
`Use of uninitialized value $_ in string eq at /usr/bin/colorgcc line 298.`
`Use of uninitialized value $_ in -x at /usr/bin/colorgcc line 299.`
so you should remove the non-existent directory in you $PATH environment variable.

:)sorry for my bad english.

**Comment by burgerdev** [comment]
@hzy199411 Thank you very much :)

**Comment by hzy199411** [comment]
@olibre, I have wrote a simple patch to fix this problem,would you like to merge it into the repo?
see https://gist.github.com/hzy199411/5e1b9cffa3a391b7630f

**Comment by olibre** [comment]
Hi @hzy199411

Thank you for having found the problem :smile: :+1: 

Your patch seems fine, please send me a *"Pull request"*.

Else, as I have just added you as collaborator on [olibre/colorgcc](https://github.com/olibre/colorgcc), you can edit directly the file colorgcc.pl in my Git repo : 
https://github.com/olibre/colorgcc/blob/master/colorgcc.pl

Have fun, cheers, 
Olibre

**Comment by olibre** [comment]
Thank you very much @hzy199411 for your fix.
I have adding more files and released version 1.4.4.
See https://github.com/olibre/colorgcc/releases/tag/1.4.4
Please @burgerdev  and @gagern Could you also check if everything is OK?

**Comment by burgerdev** [comment]
@olibre everything working as expected, thank you.
